static/js/topic_list.js

```diff
@@ -159,7 +178,41 @@ export class TopicListWidget {
         return this.my_stream_id;
     }
 
+    update_topic_search_text() {
+        const input = this.parent_elem.find("#filter-topic-input");
+        if (input.length) {
+            this.topic_search_text = input.val();
+        } else {
+            this.topic_search_text = "";
+        }
+    }
```

Comment:
I think this might have a cleaner interface if it were `save_topic_search_text(text)`, and have the caller pass `""` or the input value.

Reply:
Yeah made these changes!

Reply:
I think passing `""` is a lot cleaner than allowing the function to be passed nothing, so I changed that.